kolibri/content/api.py

```diff
@@ -307,8 +310,18 @@ def retrieve(self, request, pk):
 
         return response
 
+    @detail_route(methods=['get'])
+    def filesizes(self, request, pk):
+        instance = self.get_object()
+
+        files = models.LocalFile.objects.filter(files__contentnode__in=instance.get_descendants(include_self=True)).distinct()
+        total_file_size = files.aggregate(Sum('file_size'))['file_size__sum'] or 0
+        on_device_file_size = files.filter(available=True).aggregate(Sum('file_size'))['file_size__sum'] or 0
+
+        return Response({'total_file_size': total_file_size, 'on_device_file_size': on_device_file_size})
+
     def _get_parent_and_children_info(self, pk, available=None):
-        queryset = self.get_queryset(available)
+        queryset = self.get_queryset(available, prefetch=True)
```

Comment:
I see that it's set here, but is prefetching the LocalFiles necessary anymore? I don't think it will get used by the queries we're doing in filesizes now.

Reply:
It's not used for getting file sizes anymore, but it's used in the `kolibri/content/serializers.py` to get each local file's file path and check whether a resource is available on the external drive when doing a local import. The code is here: https://github.com/learningequality/kolibri/blob/develop/kolibri/content/serializers.py#L311. Maybe there's a better way to do it without prefetching the Files and LocalFiles?

Reply:
Ok, got it, sounds good